### PR TITLE
Exception to the newline requirement on the first member var after class opener.

### PR DESCRIPTION
**Pull request by photodude**
Based on the draft PHPCS2 Joomla Custom member var sniff https://github.com/photodude/coding-standards/blob/phpcs-2/Joomla/Sniffs/WhiteSpace/MemberVarSpacingSniff.php#L102-L137

Addresses issues #920 and #725

### CODE REVIEW

File: CodeSniffer/Standards/Squiz/Sniffs/WhiteSpace/MemberVarSpacingSniff.php [hunk 1]
```diff
@@ -34,6 +34,16 @@ class Squiz_Sniffs_WhiteSpace_MemberVarSpacingSniff extends PHP_CodeSniffer_Stan
 
 
     /**
+     * The exception to the newline requirement on the first member var after class opener.
+     * false for newline requirement on the first member var after class opener
+     * true for no newline requirement on the first member var after class opener
+     *
+     * @var bool
+     */
+    public $firstVarException = false;
```
Comment (aik099) [inline CodeSniffer/Standards/Squiz/Sniffs/WhiteSpace/MemberVarSpacingSniff.php]: This is a BC break.

Comment (photodude) [inline CodeSniffer/Standards/Squiz/Sniffs/WhiteSpace/MemberVarSpacingSniff.php]: Maybe I'm not seeing what you're seeing, How is this a BC break?

What I see is a new API end point adding additional functionality which is controlled by a new member var parameter. 

The existing behavior for this sniff is unchanged... unless you set the new member var parameter to true thus turning on the exception.

Comment (aik099) [inline CodeSniffer/Standards/Squiz/Sniffs/WhiteSpace/MemberVarSpacingSniff.php]: Ah, I thought that you've wrapped existing code with an IF, which was disabled by default. Now I see that it's completely new code (maybe fixing part is copy-pasted from somewhere).

Comment (photodude) [inline CodeSniffer/Standards/Squiz/Sniffs/WhiteSpace/MemberVarSpacingSniff.php]: Yes, it's listed in the PR description.

**Comment by photodude** [comment]
@gsherwood can this be considered for merge?

**Comment by gsherwood** [comment]
Sorry, just getting around to look at this PR.

If options are going to be added to this sniff, I think it is better to have 2 new options; 1 to specify how many blank lines should be between member vars, and a second that specifies how many blank lines before the first member var.

The same set of options could probably be added to Squiz.WhiteSpace.FunctionSpacing. This sniff actually has a `spacing` property already to define how many blank lines should be between functions, but it doesn't have an option to say how many blank lines should be before the first function found in a file or class.

If you don't have time to look into those two new options, I'll get around to this eventually.

**Comment by photodude** [comment]
Sounds like a plan. I'm not sure when, or if, I'll get around to looking into making the modifications you suggested.

**Comment by wilsonge** [comment]
See https://github.com/squizlabs/PHP_CodeSniffer/pull/1355 for an attempted implementation

**Comment by photodude** [comment]
Closing in favor of #1355 (thanks for making the requested modifications @wilsonge )

**Comment by wilsonge** [comment]
No stress :)

**Pull request closed without merging**